### PR TITLE
Add item name translation

### PR DESCRIPTION
**Pull request by rtm516**
Fixes: https://github.com/GeyserMC/Geyser/issues/555
Fixes: https://github.com/GeyserMC/Geyser/issues/508

### CODE REVIEW

**Comment by lukeeey** [comment]
I think the item translator needs some rewriting... I did it like this 

```java
 @Override
    public CompoundTag translateToBedrock(com.github.steveice10.opennbt.tag.builtin.CompoundTag javaTag) {
        CompoundTagBuilder root = CompoundTagBuilder.builder();

        if(!javaTag.contains("display")) {
            CompoundTagBuilder display = CompoundTagBuilder.builder();

            if (javaTag.contains("name")) {
                display.stringTag("Name", Message.fromString((String) javaTag.get("name").getValue()).getFullText());
                javaTag.remove("name");
            }
            if (javaTag.contains("lore")) {
                com.nukkitx.nbt.tag.ListTag list = (com.nukkitx.nbt.tag.ListTag) ItemTranslatorRegistry.translateRawNBT((Tag) javaTag.get("lore"));
                // noinspection unchecked
                display.listTag("Lore", com.nukkitx.nbt.tag.StringTag.class, list.getValue());
                javaTag.remove("lore");
            }
            root.tag(display.build("display"));
        }
        return root.buildRootTag();
    }
```

**Comment by DoctorMacc** [comment]
Oh, absolutely, it needs rewriting. I'm not quite sure how it's done right now.

File: connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java [hunk 1]
```diff
@@ -125,12 +132,37 @@ public ItemData translateToBedrock(GeyserSession session, ItemStack stack) {
...
+                displayBuilder.stringTag("Name", name);
+
+                // Build the new root tag
+                CompoundTagBuilder builder = itemData.getTag().toBuilder();
+                builder.tag(displayBuilder.build("display"));
+
+                // Create a new item with the original data + updated name
+                itemData = ItemData.of(itemData.getId(), itemData.getDamage(), itemData.getCount(), builder.buildRootTag());
+            }
+        } catch (Exception e) { }
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java:163]: Why an all-purpose "catch exception"?

Comment (rtm516) [reply connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java:163]: It was left in, from some other code. But I thought it may need it for the name getting if the nbt is null `itemData.getTag()` will return an NPE.

Comment (DoctorMacc) [reply connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java:163]: It would likely be cleaner and more self-explanatory for test for an NPE.

File: connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java [hunk 2]
```diff
@@ -105,6 +111,7 @@ public ItemStack translateToJava(GeyserSession session, ItemData data) {
                 }
             }
         }
+
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java:114]: Remove unnecessary whitespace.

File: connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java (hunk 1 above)
Comment (DoctorMacc) [reply connector/src/main/java/org/geysermc/connector/network/translators/item/ItemTranslator.java:163]: Or, at the least, add a debug ;ine here.
